### PR TITLE
CLUS-3686 register redis_sharded cluster operation

### PR DESCRIPTION
**Pull request by alvaro-vinuela**

### CODE REVIEW

File: libs9s/s9srpcclient.cpp [hunk 1]
```diff
@@ -11849,6 +11853,94 @@ S9sRpcClient::syncClusters()
...
+    //
+    // The jobspec describing the command.
+    //
+    jobSpec["command"]          = "add_cluster";
+    jobSpec["job_data"]         = jobData;
+
+    //
+    // The job instance describing how the job will be executed.
+    //
+    job["title"]                = "Register Redis Cluster";
```
Comment (Piotr-S9s) [inline libs9s/s9srpcclient.cpp:11932]: '"Register Redis **Sharded** Cluster"' ?

File: libs9s/s9srpcclient.cpp [hunk 2]
```diff
@@ -4533,7 +4537,14 @@ S9sRpcClient::createRedisSharded(
         jobData["redis_sharded_port"] = options->redisShardedPort();
 
     if (options->redisShardedBusPort() != 0)
+    {
+        if(version.contains("6"))
```
Comment (Piotr-S9s) [inline libs9s/s9srpcclient.cpp:4541]: What if there will be redis 7.6 or 7.4.6? Please change this condition to verify major number only

File: libs9s/s9srpcclient.cpp [hunk 3]
```diff
@@ -11849,6 +11860,93 @@ S9sRpcClient::syncClusters()
...
+    if (hosts.size() < 1u)
+    {
+        PRINT_ERROR(
+                "Node is not specified while registering existing cluster.");
+        return false;
+    }
+    else if (hosts.size() > 1u)
+    {
+
+        PRINT_ERROR("Redis sharded clusters requires only one cluster's node to discover topology");
```
Comment (Piotr-S9s) [inline libs9s/s9srpcclient.cpp:11885]: If it requires only one node, have you considered to take the first one and proceed?

File: libs9s/s9srpcclient.cpp [hunk 4]
```diff
@@ -11849,6 +11860,93 @@ S9sRpcClient::syncClusters()
...
+
+    //
+    // Required parameters
+    //
+    if(options->redisShardedPort() != 0)
+        jobData["redis_sharded_port"] = options->redisShardedPort();
+    else
+    {
+        PRINT_ERROR(
+                "Redis sharded clusters requires '--redis-sharded-port' option");
```
Comment (Piotr-S9s) [inline libs9s/s9srpcclient.cpp:11911]: typo -> `cluster requires`  (I think)

**Review by Piotr-S9s — CHANGES_REQUESTED** [review]

File: libs9s/s9srpcclient.cpp (hunk 3 above)
Comment (alvaro-vinuela) [reply libs9s/s9srpcclient.cpp:11885]: I think it is more clear the command usaga this way

File: libs9s/s9soptions.cpp [hunk 5]
```diff
@@ -2179,6 +2180,18 @@ S9sOptions::redisNodeTimeoutMs() const
     return 0;
 }
 
+/**
+ * \returns The value of the 'cluster-replica-validity-factor- to use on configuration
```
Comment (Piotr-S9s) [inline libs9s/s9soptions.cpp:2184]:
```suggestion
 * \returns The value of the 'cluster-replica-validity-factor' to use on configuration
```

File: libs9s/s9srpcclient.cpp [hunk 6]
```diff
@@ -4533,10 +4537,25 @@ S9sRpcClient::createRedisSharded(
         jobData["redis_sharded_port"] = options->redisShardedPort();
 
     if (options->redisShardedBusPort() != 0)
+    {
+        int versionNum = 0;
+        if(!version.empty())
+        {
+           char majorVersion = version.c_str()[0];
```
Comment (Piotr-S9s) [inline libs9s/s9srpcclient.cpp:4544]:
```suggestion
           if(version.c_str()[0] == '6')
           {
                PRINT_ERROR("Redis 6 does not support redis sharded bus port");
                return false;
           }
```

**Review by Piotr-S9s — APPROVED** [review]

**Pull request merged**